warehouse/packaging/models.py

```diff
@@ -717,7 +718,12 @@ def uploaded_via_trusted_publisher(self) -> bool:
         """Return True if the file was uploaded via a trusted publisher."""
         return (
             self.events.where(
-                self.Event.additional.op("->>")("publisher_url").is_not(None)  # type: ignore[attr-defined] # noqa E501
+                or_(
+                    self.Event.additional[  # type: ignore[attr-defined]
+                        "uploaded_via_trusted_publisher"
+                    ].as_boolean(),  # noqa E501
+                    self.Event.additional.op("->>")("publisher_url").is_not(None),  # type: ignore[attr-defined] # noqa E501
```

Comment:
I wonder if this should follow the `as_*()` approach, since then we're not asking for PG-specific operations, and following the new pattern for consistency.

```suggestion
                    self.Event.additional["publisher_url"].as_string().is_not(None)  # type: ignore[attr-defined] # noqa E501
```

Reply:
Makes sense to me! I was having trouble with this one for brainfart reasons.